lib/spack/spack/cmd/stage.py

```diff
@@ -50,4 +50,6 @@ def stage(parser, args):
     specs = spack.cmd.parse_specs(args.specs, concretize=True)
     for spec in specs:
         package = spack.repo.get(spec)
-        package.do_stage()
+        with package.stage as stage:
+            stage.delete_on_exit = False
+            package.do_stage()
```

Comment:
@tgamblin The three lines here went in accidentally. They are not wrong, but I didn't mean to put them in before we agreed whether or not it is better to remove the explicit call to `stage.create()` in `Package._make_stage()`. Just tell me if you want these reverted.